pkg/cluster/cluster.go

```diff
@@ -495,13 +495,14 @@ func (c *Cluster) updateTPRStatus() error {
 		return nil
 	}
 
-	newCluster := c.cluster
-	newCluster.Status = c.status
-	newCluster, err := k8sutil.UpdateClusterTPRObject(c.config.KubeCli.Core().RESTClient(), c.cluster.Metadata.Namespace, newCluster)
+	updateFunc := func(cl *spec.Cluster) {
+		cl.Spec = c.cluster.Spec
+		cl.Status = c.status
+	}
+	newCluster, err := k8sutil.AtomicUpdateClusterTPRObject(c.config.KubeCli.Core().RESTClient(), c.cluster.Metadata.Name, c.cluster.Metadata.Namespace, updateFunc)
```

Comment:
`updateTPRStatus()` is being retried in another method, `reportFailedStatus()` IIRC.

We need further thoughts on how to change existing `UpdateClusterTPRObject` before changing it.

Let's make this PR change only the test files, and deal with other places later in separate PR.